ui/src/components/canvas/FlowCanvas.tsx

```diff
@@ -151,6 +180,12 @@ const FlowCanvas = () => {
             <ControlButton title="change density" onClick={updateLayoutDensity}>
               <Maximize />
             </ControlButton>
+            <ControlButton title="undo" onClick={() => undo}>
+              <Undo />
+            </ControlButton>
+            <ControlButton title="redo" onClick={() => redo}>
```

Comment:
Looks like some parens were left off after the recent changes, should these be `onClick={() => undo()}` and `onClick={() => redo()}`?

Reply:
I just removed the parenthesis as it looked cleaner